SECP-service/src/main/java/com/visucius/secp/SECPService.java

```diff
@@ -58,9 +59,10 @@ public void run(SECPConfiguration configuration,
                     Environment environment) throws Exception {
         environment.jersey().setUrlPattern("/SECP/*");
 
+        final UserDAO userDAO =  new UserDAO(hibernateBundle.getSessionFactory());
+        final UserRegistrationController userRegistrationController = new UserRegistrationController(userDAO);
 
-
-        environment.jersey().register(new UserResource(
-            new UserDAO(hibernateBundle.getSessionFactory())));
+        environment.jersey().register(
+            new UserResource(userDAO,userRegistrationController));
```

Comment:
if the controller already has a dao, i don't see a point of having dao in the constructor of the resource